src/redux/slices/persistGateHydration.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { REHYDRATE } from 'redux-persist';
+
+import { RootState } from '../RootState';
+
+import SliceName from '@/redux/types/SliceName';
+
+export type PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: boolean;
+  isPersistGateHydrationComplete: boolean;
+};
+
+const initialState: PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: false,
+  isPersistGateHydrationComplete: false,
+};
+
+// This slice checks if redux-persist has finished hydrating the store.
+// Hydration happens a bit after the page loads, useful for preventing
+// the animations until the state is settled in.
+export const persistGateHydrationSlice = createSlice({
+  name: SliceName.PERSIST_GATE_HYDRATION,
+  initialState,
+  reducers: {
+    setIsPersistGateHydrationInProgress: (
+      state: PersistGateHydrationState,
+      action: PayloadAction<boolean>,
+    ) => ({
+      ...state,
+      isPersistGateHydrationInProgress: action.payload,
+    }),
+    setIsPersistGateHydrationComplete: (
+      state: PersistGateHydrationState,
+      action: PayloadAction<boolean>,
+    ) => ({
+      ...state,
+      isPersistGateHydrationComplete: action.payload,
+    }),
+  },
+  extraReducers: (builder) => {
+    builder.addCase(REHYDRATE, (state: PersistGateHydrationState) => {
+      // eslint-disable-next-line no-param-reassign
+      state.isPersistGateHydrationInProgress = true;
+    });
+  },
+});
+
+export const { setIsPersistGateHydrationInProgress, setIsPersistGateHydrationComplete } =
+  persistGateHydrationSlice.actions;
+
+export const selectisPersistGateHydrationInProgress = (state: RootState) =>
+  state.persistGateHydration.isPersistGateHydrationInProgress;
+export const selectisPersistGateHydrationComplete = (state: RootState) =>
```

Comment:
nit: `selectIsPersistGateHydrationComplete` and `selectIsPersistGateHydrationInProgress`.